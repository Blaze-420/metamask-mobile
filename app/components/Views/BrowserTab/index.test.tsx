import React from 'react';
import { shallow } from 'enzyme';
import { BrowserTab } from './';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { backgroundState } from '../../../util/test/initial-root-state';
import { MOCK_ACCOUNTS_CONTROLLER_STATE } from '../../../util/test/accountsControllerTestUtils';
const mockNavigation = {
  goBack: jest.fn(),
  goForward: jest.fn(),
  canGoBack: true,
  canGoForward: true,
};
const mockInitialState = {
  browser: { activeTab: '' },
  engine: {
    backgroundState: {
      ...backgroundState,
      AccountsController: MOCK_ACCOUNTS_CONTROLLER_STATE,
    },
  },
  transaction: {
    selectedAsset: '',
  },
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockImplementation(() => mockInitialState),
}));

const mockStore = configureMockStore();
const store = mockStore(mockInitialState);

const mockProps = {
  id: 1,
  activeTab: 1,
  defaultProtocol: 'https://',
  ipfsGateway: 'https://gateway.ipfs.io/ipfs/',
  selectedAddress: '0x123',
  whitelist: [],
  bookmarks: [],
  searchEngine: 'Google',
  newTab: jest.fn(),
  addBookmark: jest.fn(),
  addToBrowserHistory: jest.fn(),
  addToWhitelist: jest.fn(),
  updateTabInfo: jest.fn(),
  showTabs: jest.fn(),
  setOnboardingWizardStep: jest.fn(),
  wizardStep: 1,
  isIpfsGatewayEnabled: false,
  chainId: '1',
  isInTabsView: false,
};

describe('Browser', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <BrowserTab
          initialUrl="https://metamask.io"
          {...mockProps}
          navigation={mockNavigation}
        />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
