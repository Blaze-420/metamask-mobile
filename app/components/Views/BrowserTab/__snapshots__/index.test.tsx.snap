// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserTab should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
      },
      {
        "display": "none",
        "flex": 0,
        "height": 0,
        "opacity": 0,
        "width": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      {
        "height": 3,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
        "zIndex": 999999,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "opacity": 1,
        }
      }
      visible={true}
    >
      <View
        animated={true}
        animationConfig={
          {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={false}
        onLayout={[Function]}
        progress={0}
        style={
          [
            {
              "backgroundColor": undefined,
              "borderColor": "#0376c9",
              "borderRadius": 0,
              "borderWidth": 0,
              "overflow": "hidden",
              "width": null,
            },
            undefined,
          ]
        }
        useNativeDriver={true}
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#0376c9",
              "height": 3,
              "transform": [
                {
                  "translateX": -0,
                },
                {
                  "translateX": -0,
                },
                {
                  "scaleX": 0.0001,
                },
              ],
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
